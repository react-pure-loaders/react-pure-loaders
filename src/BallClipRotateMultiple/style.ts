import { keyframes } from '@emotion/react';
import { globalAnimation } from '../mixins';
import { PRIMARY_COLOR } from '../variables';

const rotate = keyframes({
  '0%': {
    transform: 'rotate(0deg) scale(1)',
  },
  '100%': {
    transform: 'rotate(360deg) scale(1)',
  },
  '50%': {
    transform: 'rotate(180deg) scale(0.6)',
  },
});

export default (color?: string) => ({
  position: 'relative' as 'relative',
  ' > div': {
    ...globalAnimation(),
    animation: `${rotate} 1s 0s ease-in-out infinite`,
    border: `2px solid ${color || PRIMARY_COLOR}`,
    borderBottomColor: 'transparent',
    borderRadius: '100%',
    borderTopColor: 'transparent',
    height: '35px',
    left: '-20px',
    position: 'absolute' as 'absolute',
    top: '-20px',
    width: '35px',

    '&:last-child': {
      animationDirection: 'reverse',
      animationDuration: '0.5s',
      borderColor: `${color || PRIMARY_COLOR} transparent ${color
        || PRIMARY_COLOR} transparent`,
      display: 'inline-block',
      height: '15px',
      left: '-10px',
      top: '-10px',
      width: '15px',
    },
  },
});
