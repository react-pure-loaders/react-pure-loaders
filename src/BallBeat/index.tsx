/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallBeat({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
