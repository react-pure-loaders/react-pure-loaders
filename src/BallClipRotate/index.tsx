/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallClipRotate({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)} data-testid="ball-clip-rotate-container">
                <div/>
            </div>
    );
  }

  return null;
}
