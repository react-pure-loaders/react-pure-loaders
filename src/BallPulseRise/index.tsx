/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallPulseRise({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
                <div/>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
