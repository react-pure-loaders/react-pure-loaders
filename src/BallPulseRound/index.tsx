/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallPulseRound({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
            </div>
    );
  }

  return null;
}
