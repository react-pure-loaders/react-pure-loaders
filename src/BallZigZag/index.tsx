/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallZigZag({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
