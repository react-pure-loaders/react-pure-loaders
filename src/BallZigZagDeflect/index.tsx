/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallZigZagDeflect({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
