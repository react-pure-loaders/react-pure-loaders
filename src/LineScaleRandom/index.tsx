/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
export default function LineScaleRandom({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div />
                <div />
                <div />
                <div />
            </div>
    );
  }

  return null;
}
