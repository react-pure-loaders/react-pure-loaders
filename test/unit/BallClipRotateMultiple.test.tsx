import React from 'react';
import Chance from 'chance';
import { matchers } from '@emotion/jest';
import { render, cleanup } from '@testing-library/react';


import BallClipRotateMultiple from '../../src/BallClipRotateMultiple';
import { PRIMARY_COLOR } from '../../src/variables';

expect.extend(matchers);

const chance = new Chance();

describe('<BallClipRotateMultiple>', () => {
    afterEach(cleanup);

    test('BallClipRotateMultiple should match snapshot', () => {
        const { container } = render(<BallClipRotateMultiple loading={true}/>);

        expect(container.firstChild).toMatchSnapshot();
    });

    test.skip('BallClipRotateMultiple should have default color', () => {
        const { container } = render(<BallClipRotateMultiple loading={true}/>);

        expect(container.firstChild).toHaveStyleRule(
            'border',
            `2px solid ${PRIMARY_COLOR}`,
            {
                target: '> div',
            },
        );
        // TODO
        // expect(container.firstChild).toHaveStyleRule(
        //     'border-color',
        //     `${PRIMARY_COLOR} transparent ${PRIMARY_COLOR} transparent`,
        //     {
        //         target: '> div &:last-child',
        //     },
        // );
    });

    test.skip('BallClipRotateMultiple should have given color', () => {
        const color = chance.color({ format: 'hex' });
        const { container } = render(<BallClipRotateMultiple color={color} loading={true}/>);

        expect(container.firstChild).toHaveStyleRule(
            'border',
            `2px solid ${color}`,
            {
                target: '> div',
            },
        );
        // TODO
        // expect(container.firstChild).toHaveStyleRule(
        //     'border-color',
        //     `${color} transparent ${color} transparent`,
        //     {
        //         target: '> div &:last-child',
        //     },
        // );
    });

    test('BallClipRotateMultiple should have no children', () => {
        const { container } = render(<BallClipRotateMultiple loading={false}/>);

        expect(container.firstChild).toBeNull();
    });

    test('BallClipRotateMultiple should have three children', () => {
        const { container } = render(<BallClipRotateMultiple loading={true}/>);

        expect(container.querySelectorAll('div')).toHaveLength(3);
    });
});
