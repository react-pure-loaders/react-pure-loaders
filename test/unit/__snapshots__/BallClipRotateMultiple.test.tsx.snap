// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BallClipRotateMultiple> BallClipRotateMultiple should match snapshot 1`] = `
@keyframes animation-0 {
  0% {
    -webkit-transform: rotate(0deg) scale(1);
    -moz-transform: rotate(0deg) scale(1);
    -ms-transform: rotate(0deg) scale(1);
    transform: rotate(0deg) scale(1);
  }

  100% {
    -webkit-transform: rotate(360deg) scale(1);
    -moz-transform: rotate(360deg) scale(1);
    -ms-transform: rotate(360deg) scale(1);
    transform: rotate(360deg) scale(1);
  }

  50% {
    -webkit-transform: rotate(180deg) scale(0.6);
    -moz-transform: rotate(180deg) scale(0.6);
    -ms-transform: rotate(180deg) scale(0.6);
    transform: rotate(180deg) scale(0.6);
  }
}

.emotion-0 {
  position: relative;
}

.emotion-0>div {
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
  -webkit-animation: animation-0 1s 0s ease-in-out infinite;
  animation: animation-0 1s 0s ease-in-out infinite;
  border: 2px solid #ffffff;
  border-bottom-color: transparent;
  border-radius: 100%;
  border-top-color: transparent;
  height: 35px;
  left: -20px;
  position: absolute;
  top: -20px;
  width: 35px;
}

.emotion-0>div:last-child {
  -webkit-animation-direction: reverse;
  animation-direction: reverse;
  -webkit-animation-duration: 0.5s;
  animation-duration: 0.5s;
  border-color: #ffffff transparent #ffffff transparent;
  display: inline-block;
  height: 15px;
  left: -10px;
  top: -10px;
  width: 15px;
}

<div
  class="emotion-0"
>
  <div />
  <div />
</div>
`;
